### PR TITLE
feat(openapi-parser): migrate openapi parser to monorepo

### PR DESCRIPTION
**Pull request by tmastrom**
This PR migrates the `openapi-parser` package to the monorepo

Real API specifications previously fetched from apis.guru and `.yaml` and `.json` testing specifications have been moved to their own Google Cloud Storage bucket `test-specifications` and are fetched on an as needed basis for testing.

### CODE REVIEW

File: packages/openapi-parser/pnpm-lock.yaml
Comment (marclave) [inline packages/openapi-parser/pnpm-lock.yaml:1]: @tmastrom should there be a lock file in here?

File: packages/openapi-parser/src/types/openapi-types.ts
Comment (marclave) [inline packages/openapi-parser/src/types/openapi-types.ts:1]: @amritk @tmastrom did we recently move types around with our new types pacakge ?

Comment (amritk) [reply packages/openapi-parser/src/types/openapi-types.ts:1]: yea its been extracted, we can point this to those but we can do it in a different PR

File: packages/openapi-parser/pnpm-lock.yaml
Comment (amritk) [reply packages/openapi-parser/pnpm-lock.yaml:1]: delete it!

**Review by geoffgscott — APPROVED** [review]
Looks good.

**Review by hanspagel — CHANGES_REQUESTED** [review]
Pls give me a chance to review before merging 🙏

**Pull request closed without merging**